src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -83,11 +84,15 @@ public FindCommand parse(String args) throws ParseException {
         }
 
         if (hasAddressPrefix) {
-            String addressInput = argMultimap.getValue(PREFIX_ADDRESS).get().trim(); // Get the actual address input
-            if (addressInput.isEmpty()) {
-                throw new ParseException("Address cannot be empty!");
+            // Collect all address inputs
+            List<String> addressKeywords = new ArrayList<>();
+            for (String address : argMultimap.getAllValues(PREFIX_ADDRESS)) {
+                String addressInput = address.trim();
+                if (addressInput.isEmpty()) {
+                    throw new ParseException("Address cannot be empty!");
```

Comment:
I think it would be good to abstract out the exception message

Reply:
Yep thanks for pointing out, will make the change